tests/infrastructure/tekton/utils.py

```diff
@@ -94,3 +106,25 @@ def process_yaml_files(file_paths, replacements, resource_kind, namespace):
         update_tekton_resources_yaml_file(file_path=file_path, replacements=replacements)
         resources.append(resource_kind(yaml_file=file_path, namespace=namespace).create())
     return resources
+
+
+def wait_and_assert_pipelinerun_succeeded(
+    pipelinerun: PipelineRun, wait_timeout: int = TIMEOUT_50MIN, sleep_interval: int = TIMEOUT_1MIN
+) -> None:
+    try:
+        for sample in TimeoutSampler(
+            wait_timeout=wait_timeout,
+            sleep=sleep_interval,
+            func=lambda: pipelinerun.instance.status.conditions[0],
+        ):
+            if sample.status != Resource.Condition.Status.UNKNOWN:
+                LOGGER.info(f"PipelineRun Condition : {sample}")
+                break
+
+        assert sample.status == Resource.Condition.Status.TRUE and sample.type == Resource.Condition.Phase.SUCCEEDED, (
```

Comment:
I would split the assert part out of this function. It should return sample when the condition is met.

Reply:
Based on https://github.com/RedHatQE/openshift-virtualization-tests/pull/81#discussion_r1920458086 , I felt this would e a right thing to do.